plugins/hypervisors/kvm/src/main/java/com/cloud/hypervisor/kvm/resource/LibvirtVMDef.java

```diff
@@ -1421,13 +1438,18 @@ public String getContent() {
                 netBuilder.append(_dpdkExtraLines);
             }
 
-            if (_netType != GuestNetType.VHOSTUSER) {
+            if (_netType != GuestNetType.VHOSTUSER && _netType != GuestNetType.USER) {
                 netBuilder.append("<link state='" + (_linkStateUp ? "up" : "down") +"'/>\n");
             }
 
             if (_slot  != null) {
                 netBuilder.append(String.format("<address type='pci' domain='0x0000' bus='0x00' slot='0x%02x' function='0x0'/>\n", _slot));
             }
+
+            if (StringUtils.isNotBlank(_userIp4Network) && _userIp4Prefix != null) {
+                netBuilder.append(String.format("<ip family='ipv4' address='%s' prefix='%s'/>\n", _userIp4Network, _userIp4Prefix));
```

Comment:
sonar advice , i think it makes sense, even though we are pretty sure we run on linux ;)
```suggestion
                netBuilder.append(String.format("<ip family='ipv4' address='%s' prefix='%s'/>%n", _userIp4Network, _userIp4Prefix));
```

Reply:
Good catch @DaanHoogland - my only reservation is that it would be different from all of the other XML formatting strings in this `getContent()` method.

Reply:
It is not serious @mlsorensen , just a trivial warning and i am just mentioning it without a request for change. We only run this on Linux AFAIK. If you want to bother go ahead, we could also create a ticket to change all occurrences of `\n` to `%n` in `String.format()` occurrences. But I know I can be over-pedantic in these matters so I tend to not push for these formalities.

Reply:
I'd probably opt to leave it as-is, rather than potentially inserting one random carriage return or something different on one line of the file compared to all of the other lines. 

I don't think a theoretical libvirt server would care/require a carriage or alternate whitespace on Windows or other OS anyway.